### ISSUE TITLE
-1001 error when connecting to docker local media server

### ISSUE THREAD

**Issue by alteredtech** [jellyfin/Swiftfin#338]
### Describe the bug

1. attempt to join media-server on local network
2. connection will grey out and loading icon will spin
3. after a few moments it will give -1001 error network timed out
4. These are the logs that are made when attempting to connect
```
[2022-01-21 12:14:55.061 -07:00] [INF] [39] Emby.Server.Implementations.HttpServer.WebSocketManager: WS "10.0.2.1" request
[2022-01-21 12:15:56.071 -07:00] [WRN] [42] Emby.Server.Implementations.HttpServer.WebSocketConnection: WS "10.0.2.1" error receiving data: "The remote party closed the WebSocket connection without completing the close handshake."
[2022-01-21 12:15:56.156 -07:00] [INF] [42] Emby.Server.Implementations.HttpServer.WebSocketManager: WS "10.0.2.1" closed
```

The 10.0.2.1 is the ip of my opnsense router.
When I use my public FQDN for jellyfin it connects without issues. Nothing shows in the logs when I click on connect.

### Application version

1.0.0 (70)

### Where did you install the app from?

TestFlight

### Device information

AppleTV 4K gen-2

### OS version

tvOS 15.2

### Jellyfin server version

10.7.7 (docker)

**Comment by LePips** [comment]
That is a timeout issue. Are you using TLS 1.3?

**Comment by alteredtech** [comment]
I don't believe so, I also put in http:// so it should not be using any TLS.

**Comment by LePips** [comment]
Sorry to say this might be almost impossible for me to debug, as I don't have this network setup.

**Comment by hendrik1120** [comment]
Empty logs are weird... If you don't mind, could you please check the following:
Can you connect locally from any other jellyfin clients on the same network as the apple tv?
Are there any firewall rules in place that might drop the packets to cause the 1001?

**Comment by alteredtech** [comment]
There are no firewall rules in place to drop the packets, anything goes on each individual lan, isolate between vlans and only main vlan can talk to the others. My client and server are on the same vlan. I tried on my phone picking the media-server local connection. It also failed.
While working I was wondering why my router was showing on the logs since it was my apple tv. I entered the ip of the server and the port, worked like a charm. So it seems that since jellyfin is reporting its ip which is the container ip, the client is trying to route there since it is not on the same subnet hence why my router was in the logs. I bet if the container used the host network rather than bridge, it would be fine.
